sky/backends/backend_utils.py

```diff
@@ -2061,8 +2061,20 @@ def validate_schema(obj, schema, err_msg_prefix=''):
             raise ValueError(err_msg)
 
 
-def is_public_cloud_disabled():
-    """Checks if none of the public clouds are enabled."""
-    enabled_clouds = global_user_state.get_enabled_clouds()
-    return len(enabled_clouds) == 1 and isinstance(enabled_clouds[0],
-                                                   clouds.Local)
+def check_public_cloud_enabled():
+    """Checks if any of the public clouds is enabled."""
+
+    def _no_public_cloud():
+        enabled_clouds = global_user_state.get_enabled_clouds()
+        return (len(enabled_clouds) == 0 or
+                (len(enabled_clouds) == 1 and
+                 isinstance(enabled_clouds[0], clouds.Local)))
+
+    if not _no_public_cloud():
+        return
+
+    sky_check.check(quiet=True)
+    if _no_public_cloud():
+        with ux_utils.print_exception_no_traceback():
+            raise RuntimeError('Cloud access is not set up. '
```

Comment:
nit: any easy way to remove RuntimeError being printed on the CLI? (We need it to be printed for the first `sky.launch()` though.) Very minor.

Reply:
Good point! Changed the behavior for CLI, by explicitly calling this method before the `sky.optimize`. Now it wil shows the following:
<img width="395" alt="image" src="https://user-images.githubusercontent.com/6753189/201496384-2e8cf935-fec9-4fff-b42d-e4306e86e24e.png">